packages/core/src/Slider/index.tsx

```diff
@@ -1,4 +1,4 @@
-import React, { useState, useEffect, useRef, useCallback } from 'react'
```

Comment:
Seems like we are adding quite a bit of complexity to this component.

How hard would it be to leave the current Slider alone and create a TickSlider instead?

Reply:
Valid point, but I think we could address it when it becomes a problem, e.g. if someone uses the simple slider a lot it could be refactored out as a separate slider variant. Even if it would be a breaking changes, only the ticked slider usage would be affected, and it would just be a name change basically.